### PR TITLE
Modifying CircleCI caching keys

### PR DESCRIPTION
**Pull request by sansb**
- Adding a version to allow us to easily "bump" and invalidate the cache
- Removing the more general keys since they are either unused or harmful even though Circle docs recommend them

Doing this per CircleCI support's recommendation (ticket #35190).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 version: 2
 jobs:
+  resource_class: xlarge
```
Comment (ferlatte) [inline .circleci/config.yml]: Is this on purpose?

Comment (ferlatte) [reply .circleci/config.yml]: ie, this doesn't appear to be related to the cache keys.

Comment (sansb) [reply .circleci/config.yml]: Accident. Removing.

Comment (sansb) [reply .circleci/config.yml]: @ferlatte To give some context on this- we were looking into increasing our instance size in order to address some flakes we've been seeing on Circle. In order to do that, you have to contact Circle support and have them enable that feature for you.

**Pull request merged**